tests/test_request_left.py

```diff
@@ -0,0 +1,60 @@
+from twisted.internet import defer
+from twisted.trial.unittest import TestCase
+from scrapy.signals import request_left_downloader
+from scrapy.spiders import Spider
+from scrapy.utils.test import get_crawler
+from tests.mockserver import MockServer
+
+
+class SignalCatcherSpider(Spider):
+    name = 'signal_catcher'
+
+    def __init__(self, crawler, url, *args, **kwargs):
+        super(SignalCatcherSpider, self).__init__(*args, **kwargs)
+        crawler.signals.connect(self.on_response_download,
+                                signal=request_left_downloader)
+        self.catched_times = 0
+        self.start_urls = [url]
+
+    @classmethod
+    def from_crawler(cls, crawler, *args, **kwargs):
+        spider = cls(crawler, *args, **kwargs)
+        return spider
+
+    def on_response_download(self, request, spider):
```

Comment:
Should this be called differently, if it will called even if the request fails and the response is not actually downloaded?

Reply:
Renamed in https://github.com/scrapy/scrapy/pull/4303/commits/4be19e443e9c101a248c21509ae8000ce500d51a